flytekit/remote/execution.py

```diff
@@ -0,0 +1,105 @@
+from dataclasses import dataclass
+from typing import Any, Dict, Union, List
+
+from flytekit import ContainerTask
+from flytekit.annotated.context_manager import SerializationSettings
+from flytekit.annotated.promise import translate_inputs_to_literals
+from flytekit.annotated.reference_entity import ReferenceEntity
+from flytekit.common.core.identifier import WorkflowExecutionIdentifier
+from flytekit.common.exceptions import user as user_exceptions
+from flytekit.common.translator import get_serializable
+from flytekit.common.workflow_execution import SdkWorkflowExecution
+from flytekit.models.common import Notification, Annotations, Labels, AuthRole
+from flytekit.models.execution import ExecutionSpec, ExecutionMetadata, NotificationList
+from flytekit.clients.friendly import SynchronousFlyteClient
+from flytekit.models.literals import LiteralMap
+from flytekit.annotated import context_manager as flyte_context
+from flytekit.models.task import TaskSpec
+from flytekit.configuration import auth as auth_config
+
+
+@dataclass
+class ExecutionParameters(object):
+    notifications: List[Notification] = None
+    labels: Labels = None
+    annotations: Annotations = None
+    auth_role: AuthRole = None
+    # optional name to assign to execution
+    name: str = None
+    # The name of the user launching this execution.
+    principal: str = None
+
+
+def execute(
+    entity: Union[ReferenceEntity, ContainerTask],
+    client: SynchronousFlyteClient,
+    project: str,
+    domain: str,
+    inputs: Dict[str, Any] = None,
+    exec_parameters: ExecutionParameters = None,
+    serialization_settings: SerializationSettings = None,
+) -> SdkWorkflowExecution:
+
+    # If serialization settings are supplied, first serialize and register the entity before attempting to launch
+    # an execution with it.
+    # If the entity already exists as is, that's fine. We will proceed to execute it regardless.
+    if isinstance(entity, ContainerTask) and serialization_settings is not None:
+        with flyte_context.FlyteContext.current_context().new_serialization_settings(
+            serialization_settings=serialization_settings
+        ) as ctx:
+            sdk_task = get_serializable(ctx.serialization_settings, entity)
+            reference_identifier = sdk_task.id
+            try:
+                client.create_task(task_identifer=sdk_task.id, task_spec=TaskSpec(sdk_task))
+            except user_exceptions.FlyteEntityAlreadyExistsException:
+                pass
+    else:
+        reference_identifier = entity.id
+
+    with flyte_context.FlyteContext.current_context():
+        literals = translate_inputs_to_literals(
+            ctx, input_kwargs=inputs, interface=entity.interface, native_input_types=entity.get_input_types()
```

Comment:
the referenceEntity may not have an interface right? so we should fetch that first?

Reply:
oh I did not know this! good call, yes

Reply:
actually it looks like inputs and outputs are "required" input arguments for reference entities but i will fetch if they are none